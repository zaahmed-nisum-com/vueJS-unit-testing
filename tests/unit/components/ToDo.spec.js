import { shallowMount, mount } from "@vue/test-utils";
import Todo from "../../../src/components/ToDo";

describe("The Todo.vue component", () => {
  test("snapshot", () => {
    const wrapper = mount(Todo);
    expect(wrapper.find("div")).toMatchSnapshot(true);
  });

  test("Displays the title when passed as a prop", () => {
    const wrapper = shallowMount(Todo, {
      propsData: {
        title: "A random title",
      },
    });
    expect(wrapper.text()).toMatch("A random title");

    const wrapper2 = shallowMount(Todo, {
      propsData: {
        title: "Another random one",
      },
    });
    expect(wrapper2.text()).toMatch("Another random one");
  });

  describe("adding todo items", () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(Todo, {
        propsData: {
          title: "My list",
        },
      });
    });

    async function addTodo(todoText) {
      wrapper.find('[data-testid="todo-input"]').setValue(todoText);
      await wrapper.find('[data-testid="todo-submit"]').trigger("click");
    }

    function elementText(testId) {
      return wrapper.find(`[data-testid="${testId}"]`).text();
    }

    test("allows for adding one todo item", async () => {
      await addTodo("My first todo item");
      expect(elementText("todos")).toContain("My first todo item");
    });

    test("allows for more than one todo item to be added", async () => {
      await addTodo("My first todo item");
      await addTodo("My second todo item");
      expect(elementText("todos")).toContain("My first todo item");
      expect(elementText("todos")).toContain("My second todo item");
    });

    test("empties the input field when todo has been added", async () => {
      await addTodo("This is not important");
      expect(wrapper.find('[data-testid="todo-input"]').element.value).toEqual(
        ""
      );
    });

    test("displays the items in the order they are entered", async () => {
      await addTodo("First");
      await addTodo("Second");
      expect(elementText("todo-0")).toMatch("First");
      expect(elementText("todo-1")).toMatch("Second");
    });
  });
});
