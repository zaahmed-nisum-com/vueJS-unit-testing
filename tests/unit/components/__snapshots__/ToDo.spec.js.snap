// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Todo.vue component snapshot 1`] = `
<div>
  <h2 class="mb-4"></h2> <input type="text" data-testid="todo-input" placeholder="Add todo item..." class="border border-gray-300 p-1 text-blue-700"> <button data-testid="todo-submit" class="px-3 py-1 text-white bg-blue-500 mb-4">
    Add
  </button>
  <ul data-testid="todos" class="text-left"></ul>
</div>
`;
